### DISCUSSION TITLE
SQL Assistant not working with query containing blank lines

### DISCUSSION THREAD

**Discussion by williambuchanan2** [dbeaver/dbeaver#13278] [Q&A]
#### System information: 
- Operating system (distribution) and version: Windows 10
- DBeaver version: 21.1.3.202107181810
- Additional extensions

#### Connection specification:
- Database name and version: SQL
- Driver name: Microsoft JDBC Driver for SQL Server
- Do you use tunnels or proxies (SSH, SOCKS, etc)? No

#### Describe the problem you're observing:
Sql Assistant not working

#### Steps to reproduce, if exist:
Edit a query. Press CTRL+Space - Sql Assistant just shows 'no proposals'.
In previous versions I have found SQL Assistant unreliable (i.e. barely working), but now it just doesn't work at all.
![Screenshot_1](https://user-images.githubusercontent.com/32165641/126262079-49f05e09-3eba-408b-8cbc-07bccf2c6e92.png)

**Comment by Matvey16** [comment]
Can you provide more steps to reproduce an issue? What is the DDL of the table(s) and what is a whole query?

**Comment by williambuchanan2** [comment]
I have discovered the problem. It doesn't seem to like blank lines in the query. If you see the query below - it works when I remove the blank line. I guess this also explains why I had intermittent trouble with it in the past.

select top 100 *
from PartitionEquipment pe 
-- blank line below

where pe.[bang!]

**Comment by LonwoLonwo** [comment]
Hello @williambuchanan2 

This is the default behavior for DBeaver SQL Editor. You can disable "Blank line is statement delimiter" setting in Preferences -> Editors -> SQL Editor -> SQL Processing

![2021-07-21 10_43_00-DBeaver Enterprise 21 1 3 - _SNOWFLAKE_SAMPLE_DATA_ Script-29](https://user-images.githubusercontent.com/45152336/126451155-58fbb762-8c8f-4e9b-9e59-2f5a74f98921.png)

**Comment by williambuchanan2** [comment]
Thanks for the update. That explains it!
